control-plane-operator/controllers/hostedcontrolplane/v2/capi_provider/component.go

```diff
@@ -35,16 +36,29 @@ func (c *CAPIProviderOptions) NeedsManagementKASAccess() bool {
 	return true
 }
 
-func NewComponent(deploymentSpec *appsv1.DeploymentSpec, platformPolicyRules []rbacv1.PolicyRule) component.ControlPlaneComponent {
+func NewComponent(deploymentSpec *appsv1.DeploymentSpec, platformPolicyRules []rbacv1.PolicyRule, platformType hyperv1.PlatformType) component.ControlPlaneComponent {
 	capi := &CAPIProviderOptions{
 		deploymentSpec:      deploymentSpec,
 		platformPolicyRules: platformPolicyRules,
+		platformType:        platformType,
 	}
 
-	return component.NewDeploymentComponent(ComponentName, capi).
+	builder := component.NewDeploymentComponent(ComponentName, capi).
 		WithAdaptFunction(capi.adaptDeployment).
 		WithPredicate(predicate).
-		InjectAvailabilityProberContainer(util.AvailabilityProberOpts{}).
+		InjectAvailabilityProberContainer(util.AvailabilityProberOpts{})
+
+	// Inject token minter for GCP platform to support Workload Identity Federation
+	if platformType == hyperv1.GCPPlatform {
+		builder = builder.InjectTokenMinterContainer(component.TokenMinterContainerOptions{
+			TokenType:               component.CloudToken,
+			ServiceAccountName:      "capi-gcp-controller-manager",
+			ServiceAccountNameSpace: "kube-system",
+			KubeconfigSecretName:    "service-network-admin-kubeconfig",
+		})
+	}
```

Comment:
this split of the builder look weird. can we move this to the end of the function and keep the base builder as it is . 
```go
builder := component.NewDeploymentComponent(ComponentName, capi).
....

if platformType == hyperv1.GCPPlatform {
// inject token minter
}

return builder.Build()
```